### PR TITLE
fix font atlas overflow

### PR DESCRIPTION
**Pull request by verzuz**
This should fix the issue of overflowing (and panicing) font_atlasses.

### CODE REVIEW

File: crates/bevy_text/src/font_atlas_set.rs [hunk 1]
```diff
@@ -67,10 +75,30 @@ impl FontAtlasSet {
...
             }
-            if font_atlas.get_char_index(character).is_none() {
+            if !font_atlas
+                .iter()
+                .any(|atlas| atlas.get_char_index(character).is_some())
+            {
                 if let Some(outlined_glyph) = scaled_font.outline_glyph(glyph.clone()) {
                     let glyph_texture = Font::get_outlined_glyph_texture(outlined_glyph);
-                    font_atlas.add_char(textures, texture_atlases, character, &glyph_texture);
+                    let add_char_to_fontatlas = |atlas: &mut FontAtlas| -> bool {
```
Comment (cart) [inline crates/bevy_text/src/font_atlas_set.rs:84]: nit: as long as it is, naming conventions dictate that this should be `add_char_to_font_atlas` :)

File: crates/bevy_text/src/font_atlas_set.rs [hunk 2]
```diff
@@ -67,10 +75,30 @@ impl FontAtlasSet {
...
+                            texture_atlases,
+                            Vec2::new(512.0, 512.0),
+                        ));
+                        if !font_atlas.last_mut().unwrap().add_char(
+                            textures,
+                            texture_atlases,
+                            character,
+                            &glyph_texture,
+                        ) {
+                            panic!("could not add character to newly created fontatlas");
```
Comment (cart) [inline crates/bevy_text/src/font_atlas_set.rs:99]: nit: lets use type name casing here: `FontAtlas`

File: crates/bevy_text/src/font_atlas_set.rs [hunk 3]
```diff
@@ -84,12 +112,18 @@ impl FontAtlasSet {
         self.font_atlases
             .get(&FloatOrd(font_size))
```
Comment (cart) [inline crates/bevy_text/src/font_atlas_set.rs:113]: We can avoid looking up the character index twice by doing this:

```rust
    pub fn get_glyph_atlas_info(&self, font_size: f32, character: char) -> Option<GlyphAtlasInfo> {
        self.font_atlases
            .get(&FloatOrd(font_size))
            .and_then(|font_atlas| {
                font_atlas
                    .iter()
                    .find_map(|atlas| {
                        atlas
                            .get_char_index(character)
                            .map(|char_index| (char_index, atlas.texture_atlas))
                    })
                    .map(|(char_index, texture_atlas)| GlyphAtlasInfo {
                        texture_atlas,
                        char_index,
                    })
            })
    }
```

**Review by cart — COMMENTED** [review]
In general this looks good to me. Just a few small nit-picks. Thanks!

File: crates/bevy_text/src/font_atlas_set.rs [hunk 4]
```diff
@@ -55,7 +57,13 @@ impl FontAtlasSet {
         let font_atlas = self
```
[the remark this answers is not in the document]
Comment (verzuz) [reply crates/bevy_text/src/font_atlas_set.rs:57]: sure - makes sense

File: crates/bevy_text/src/font_atlas_set.rs (hunk 1 above)
Comment (verzuz) [reply crates/bevy_text/src/font_atlas_set.rs:84]: agreed - although I hate putting it into a Fn ... but clippy complained :-(

File: crates/bevy_text/src/font_atlas_set.rs (hunk 2 above)
Comment (verzuz) [reply crates/bevy_text/src/font_atlas_set.rs:99]: sure thing

File: crates/bevy_text/src/font_atlas_set.rs (hunk 3 above)
Comment (verzuz) [reply crates/bevy_text/src/font_atlas_set.rs:113]: uh that is nicer .. i still get confused at some of those combinations ;)

File: crates/bevy_text/src/font_atlas_set.rs (hunk 1 above)
Comment (cart) [reply crates/bevy_text/src/font_atlas_set.rs]: bawhaha yeah sometimes clippy is overly opinionated imo.

**Comment by cart** [comment]
Awesome. This is good to go. Thanks again!

**Pull request merged**